src/Controller/TagMappingController.php

```diff
@@ -0,0 +1,216 @@
+<?php
+
+namespace Drupal\thunder_print\Controller;
+
+use Drupal\Component\Utility\SortArray;
+use Drupal\Core\Controller\ControllerBase;
+use Drupal\Core\Entity\ContentEntityTypeInterface;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpFoundation\Request;
+
+/**
+ * Class TagMappingController.
+ *
+ * @package Drupal\thunder_print\Controller
+ */
+class TagMappingController extends ControllerBase {
+
+  /**
+   * Autocomplete path.
+   *
+   * @param \Symfony\Component\HttpFoundation\Request $request
+   *   Current request.
+   *
+   * @return \Symfony\Component\HttpFoundation\JsonResponse
+   *   Response.
+   */
+  public function autocomplete(Request $request) {
+    $string = $request->query->get('q');
+
+    if (substr_count($string, '.') == 0) {
+      list($searchKeyword, $matches) = $this->matchEntityType($string);
+    }
+    elseif (substr_count($string, '.') == 1) {
+      list($searchKeyword, $matches) = $this->matchBundle($string);
+    }
+    else {
+      list($searchKeyword, $matches) = $this->matchField($string, $request->query->get('type'));
```

Comment:
The convert Targets autocomplete fails at this position, because the second parameter of matchField() is not given as an array.
Besides fixing this, please implement tests for the auto complete fields.